packages/hs-react-ui/src/components/Button/ButtonContainers.tsx

```diff
@@ -1,7 +1,96 @@
-import styled from 'styled-components';
-import { darken } from 'polished';
-import { ButtonTypes, ButtonContainer } from './Button';
+import styled, { StyledComponentBase, css, keyframes } from 'styled-components';
+import { darken, getContrast, complement } from 'polished';
 import colors from '../../enums/colors';
+import fonts from '../../enums/fonts';
+
+type StyledButtonProps = {
+  elevation: number;
+  color?: string;
```

Comment:
`textColor` here?

it's tough - I'd think that for outline and link, the border and text color would come from a `color` prop. And for "filled" we can automatically pick between white and (their dark color of choice) or black if their color of choice isn't dark enough; but that's too complicated.

It's tempting to just say "gimme a color and for `type: fill` we'll set the text color automatically"